x/tokenomics/keeper/token_logic_modules_test.go

```diff
@@ -312,50 +249,175 @@ func TestProcessTokenLogicModules_AppStakeTooLow(t *testing.T) {
+	require.Equal(t, appBurn, appOverservicedEvent.EffectiveBurn.Amount)
+	require.Less(t, appBurn.Int64(), numTokensClaimed)
+}
+
+func TestProcessTokenLogicModules_TLMGlobalMint_Valid_MintDistributionCorrect(t *testing.T) {
+	// Test Parameters
+	appInitialStake := math.NewInt(1000000)
+	supplierInitialStake := math.NewInt(1000000)
+	supplierRevShareRatios := []float32{12.5, 37.5, 50}
+	globalComputeUnitsToTokensMultiplier := uint64(1)
+	serviceComputeUnitsPerRelay := uint64(1)
+	service := prepareTestService(serviceComputeUnitsPerRelay)
+	numRelays := uint64(1000) // By supplier for application in this session
+	numTokensClaimed := float64(numRelays * serviceComputeUnitsPerRelay * globalComputeUnitsToTokensMultiplier)
+	validatorConsAddr := sample.ConsAddress()
+
+	// Prepare the keepers
+	keepers, ctx := testkeeper.NewTokenomicsModuleKeepers(t, nil, testkeeper.WithService(*service), testkeeper.WithProposerAddr(validatorConsAddr))
+	keepers.SetService(ctx, *service)
 
-	var expectedBurnEventCoin, effectiveBurnEventCoin cosmostypes.Coin
-	err = json.Unmarshal([]byte(expectedBurnAttribute[0].Value), &expectedBurnEventCoin)
+	// Set compute_units_to_tokens_multiplier to simplify expectation calculations.
+	err := keepers.Keeper.SetParams(ctx, tokenomicstypes.Params{
+		ComputeUnitsToTokensMultiplier: globalComputeUnitsToTokensMultiplier,
+	})
 	require.NoError(t, err)
-	err = json.Unmarshal([]byte(effectiveBurnAttribute[0].Value), &effectiveBurnEventCoin)
+
+	// Add a new application with non-zero app stake end balance to assert against.
+	appStake := cosmostypes.NewCoin(volatile.DenomuPOKT, appInitialStake)
+	app := apptypes.Application{
+		Address:        sample.AccAddress(),
+		Stake:          &appStake,
+		ServiceConfigs: []*sharedtypes.ApplicationServiceConfig{{Service: service}},
+	}
+	keepers.SetApplication(ctx, app)
+
+	// Prepare the supplier revenue shares
+	supplierRevShares := make([]*sharedtypes.ServiceRevenueShare, len(supplierRevShareRatios))
+	for i := range supplierRevShares {
+		shareHolderAddress := sample.AccAddress()
+		supplierRevShares[i] = &sharedtypes.ServiceRevenueShare{
+			Address:            shareHolderAddress,
+			RevSharePercentage: supplierRevShareRatios[i],
+		}
+	}
+
+	// Add a new supplier.
+	supplierStake := cosmostypes.NewCoin(volatile.DenomuPOKT, supplierInitialStake)
+	supplier := sharedtypes.Supplier{
+		// Make the first shareholder the supplier itself.
+		OwnerAddress:    supplierRevShares[0].Address,
+		OperatorAddress: supplierRevShares[0].Address,
+		Stake:           &supplierStake,
+		Services:        []*sharedtypes.SupplierServiceConfig{{Service: service, RevShare: supplierRevShares}},
+	}
+	keepers.SetSupplier(ctx, supplier)
+
+	// Prepare the claim for which the supplier did work for the application
+	claim := prepareTestClaim(numRelays, service, &app, &supplier)
+
+	// Prepare addresses
+	daoAddr := authtypes.NewModuleAddress(govtypes.ModuleName)
```

Comment:
NIT for naming consistency

```suggestion
	daoAddress := authtypes.NewModuleAddress(govtypes.ModuleName)
```